dip-template/pallets/nft-marketplace/src/lib.rs

```diff
@@ -702,16 +656,15 @@ pub mod pallet {
 		#[pallet::call_index(3)]
 		#[pallet::weight(<T as pallet::Config>::WeightInfo::buy_token())]
 		pub fn buy_token(origin: OriginFor<T>, listing_id: u32, amount: u32) -> DispatchResult {
-			let origin = ensure_signed(origin.clone())?;
+			// You have access to the DID origin now.
```

Comment:
This is the other relevant change, which gives the extrinsic access to the DID of the submitter.